app/models/type.rb

```diff
@@ -97,9 +97,18 @@ def self.enabled_in(project)
     ::Type.includes(:projects).where(projects: { id: project })
   end
 
-  def statuses
+  def statuses(include_default: false)
     return [] if new_record?
+
     @statuses ||= ::Type.statuses([id])
+    return @statuses unless include_default
+
+    default = Status.default
+    if default.nil?
+      @statuses
+    else
+      [default] + @statuses
+    end
```

Comment:
`Type.statuses([id])` will return a scope nowadays. With your change, the method will either return a scope or an array depending on the existence of a default status.

A better approach would be to merge a scope for the default status with the scope returned from `Type.statuses([id])`. That way, the method will always return a scope. While this is done, the useless memoization `@statuses ||= ::Type.statuses([id])` should be removed and the unit test in the legacy specs ported over to the current specs with it being extended to accommodate for the added parameter.

Reply:
The method has always returned a scope or array (depending on new_record?), but I agree nonetheless.